clients/admin-ui/src/features/dataset/EditDatasetDrawer.tsx

```diff
@@ -15,28 +20,46 @@ interface Props {
 }
 const EditDatasetDrawer = ({ dataset, isOpen, onClose }: Props) => {
   const [updateDataset] = useUpdateDatasetMutation();
+  const [deleteDataset] = useDeleteDatasetMutation();
+  const router = useRouter();
   const toast = useToast();
 
   const handleSubmit = async (values: Partial<Dataset>) => {
-    if (dataset) {
-      const updatedDataset = { ...dataset, ...values };
-      try {
-        const result = await updateDataset(updatedDataset);
-        // TODO: we should systematically coerce errors into their types (see fidesops)
-        if ("error" in result && "data" in result.error) {
-          if ("data" in result.error) {
-            toast(errorToastParams(result.error.data as string));
-          } else {
-            toast(errorToastParams("An unknown error occurred"));
-          }
+    const updatedDataset = { ...dataset, ...values };
+    try {
+      const result = await updateDataset(updatedDataset);
+      // TODO: we should systematically coerce errors into their types (#803)
+      if ("error" in result && "data" in result.error) {
+        if ("data" in result.error) {
+          toast(errorToastParams(result.error.data as string));
         } else {
-          toast(successToastParams("Successfully modified dataset"));
+          toast(errorToastParams("An unknown error occurred"));
         }
-      } catch (error) {
-        toast(errorToastParams(error as string));
+      } else {
+        toast(successToastParams("Successfully modified dataset"));
       }
-      onClose();
+    } catch (error) {
+      toast(errorToastParams(error as string));
     }
+    onClose();
+  };
+
+  const handleDelete = async () => {
+    const { fides_key: fidesKey } = dataset;
+    const result = await deleteDataset(fidesKey);
+    // TODO: we should systematically coerce errors into their types (#803)
+    if ("error" in result && "data" in result.error) {
+      if ("data" in result.error) {
+        toast(errorToastParams(result.error.data as string));
```

Comment:
Just noticed these `as string` calls. Calling `.toString()` is a bit safer: if it's already a string it will stay, and if it's not you avoid any unexpected behavior.